ytypes/string_type.go

```diff
@@ -26,6 +26,35 @@ import (
 
 // Refer to: https://tools.ietf.org/html/rfc6020#section-9.4.
 
+var (
+	POSIXregexCache = map[string]*regexp.Regexp{}
+	RE2regexCache   = map[string]*regexp.Regexp{}
+)
+
+// compilePattern returns the compiled regex for the given regex
+// pattern. It caches previous look-ups for faster performance.
+// Go's regexp implementation might be relatively slow compared to other
+// languages: https://github.com/golang/go/issues/11646
+func compilePattern(pattern string, isPOSIX bool) (*regexp.Regexp, error) {
```

Comment:
For discussion -- since I thought about it when reviewing. Is there ever a case where we have the same <pattern, isPOSIX> tuple, but don't have the same regexp result -- such that we need to cache based on the path instead? I don't think so - since the result of a check against that regexp (`Match`) would always be the same but I figured I'd just raise it to figure out whether there's something else we need to consider here.

Reply:
My thought is that this shouldn't a be problem even if it were the case. The behaviour should be the same even if the output is nondeterministic, right?

Reply:
I think that's right, yes.